python/lsst/ip/isr/isrFunctions.py

```diff
@@ -520,14 +520,46 @@ def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1,
                 overscanImageI = overscanImage.convertI()
         else:
             overscanImageI = overscanImage
-
-        fitType = afwMath.stringToStatisticsProperty(fitType)
-        offImage = afwMath.makeStatistics(overscanImageI, fitType, statControl).getValue()
-        overscanFit = offImage
+        if fitType in ('MEDIAN',):
+            fitTypeStats = afwMath.stringToStatisticsProperty(fitType)
+            offImage = afwMath.makeStatistics(overscanImageI, fitTypeStats, statControl).getValue()
+            overscanFit = offImage
+        elif fitType in ('MEDIAN_PER_ROW',):
+            if hasattr(overscanImageI, "getImage"):
+                biasArray = overscanImageI.getImage().getArray()
+            else:
+                biasArray = overscanImageI.getArray()
+            shortInd = numpy.argmin(biasArray.shape)
+            if shortInd == 0:
+                # Convert to some 'standard' representation to make things easier
+                biasArray = numpy.transpose(biasArray)
+
+            fitTypeStats = afwMath.stringToStatisticsProperty('MEDIAN')
+            collapsed = []
+            for row in biasArray:
+                rowMedian = afwMath.makeStatistics(row, fitTypeStats, statControl).getValue()
+                collapsed.append(rowMedian)
+            if shortInd == 0:
+                collapsed = numpy.transpose(collapsed)
+            shortDimAmpImage = int(ampImage.array.shape[0]*ampImage.array.shape[1]/len(collapsed))
+            collapsedReshaped = numpy.repeat(collapsed, shortDimAmpImage).reshape(ampImage.array.shape)
+            offImage = afwImage.ImageF(ampImage.getDimensions())
+            overscanFit = afwImage.ImageF(overscanImageI.getDimensions())
+            if shortInd == 0:
+                offArray = numpy.transpose(offImage.getArray())
+                ovFitArray = numpy.transpose(overscanFit.getArray())
+            else:
+                offArray = offImage.getArray()
+                ovFitArray = overscanFit.getArray()
+            offArray[:, :] = collapsedReshaped
+            ovFitArray[:, :] = numpy.repeat(collapsed, biasArray.shape[1]).reshape(biasArray.shape)
```

Comment:
Is it possible to refactor this to use the simpler `offArray[:, :] = vector[:, numpy.newaxis] or vector[numpy.newaxis, :]` as is done for the spline/polynomial fits below (L629 on master)?  I think all these transposes and repeats are correct, but I find the newaxis notation easier to follow.

Reply:
Yes. I followed this suggestion (thanks), and have followed what is done in the spline/polynomial part.